### PR TITLE
feat(mongodb): add configurable option to override legacy protocol usage

### PR DESCRIPTION
**Pull request by thalesmg**
Fixes https://emqx.atlassian.net/browse/EMQX-10750

Fixes https://github.com/emqx/emqx/discussions/11428

See https://github.com/emqx/mongodb-erlang/pull/39

### <samp>🤖 Generated by Copilot at af90f82</samp>

Added support for different MongoDB protocol versions in the emqx_mongodb and emqx_bridge_mongodb applications. Updated the dependency, test case, configuration, and i18n files accordingly. Bumped the versions of the affected applications.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [na] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [na] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [x] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [na] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [na] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: rel/i18n/emqx_mongodb.hocon [hunk 1]
```diff
@@ -149,4 +149,10 @@ wait_queue_timeout.desc:
 wait_queue_timeout.label:
 """Wait Queue Timeout"""
 
+use_legacy_protocol.desc:
+"""Whether to use MongoDB's legacy protocol for communicating with the database.  The default is to attempt to automatically determine if the newer protocol."""
```
Comment (ieQu1) [inline rel/i18n/emqx_mongodb.hocon:153]:
```suggestion
"""Whether to use MongoDB's legacy protocol for communicating with the database.  The default is to attempt to automatically determine if the newer protocol is used."""
```
?

**Review by ieQu1 — APPROVED** [review]

File: rel/i18n/emqx_mongodb.hocon (hunk 1 above)
Comment (thalesmg) [reply rel/i18n/emqx_mongodb.hocon]: Oops, thanks.

File: apps/emqx_bridge_mongodb/test/emqx_bridge_mongodb_SUITE.erl [hunk 2]
```diff
@@ -494,3 +517,30 @@ t_get_status_server_selection_too_short(Config) ->
...
+    ?retry(
+        _Interval0 = 200,
+        _NAttempts0 = 20,
+        ?assertMatch({ok, connected}, emqx_resource_manager:health_check(ResourceID))
+    ),
+    WorkerPids0 = get_worker_pids(Config),
+    Expected0 = maps:from_keys(WorkerPids0, true),
+    LegacyOptions0 = maps:from_list([{Pid, mc_utils:use_legacy_protocol(Pid)} || Pid <- WorkerPids0]),
+    ?assertEqual(Expected0, LegacyOptions0),
+    {ok, _} = delete_bridge(Config),
```
Comment (paulozulato) [inline apps/emqx_bridge_mongodb/test/emqx_bridge_mongodb_SUITE.erl:533]: nit: maybe put it in a fun?

**Review by paulozulato — APPROVED** [review]

File: rel/i18n/emqx_mongodb.hocon [hunk 3]
```diff
@@ -149,4 +149,10 @@ wait_queue_timeout.desc:
 wait_queue_timeout.label:
 """Wait Queue Timeout"""
 
+use_legacy_protocol.desc:
+"""Whether to use MongoDB's legacy protocol for communicating with the database.  The default is to attempt to automatically determine if the newer protocol is supported."""
```
Comment (lafirest) [inline rel/i18n/emqx_mongodb.hocon:153]: nit: Two spaces

**Review by lafirest — APPROVED** [review]

File: rel/i18n/emqx_mongodb.hocon (hunk 3 above)
Comment (thalesmg) [reply rel/i18n/emqx_mongodb.hocon:153]: See: https://github.com/emqx/emqx/pull/11403#discussion_r1291271350

File: apps/emqx_bridge_mongodb/test/emqx_bridge_mongodb_SUITE.erl (hunk 2 above)
Comment (thalesmg) [reply apps/emqx_bridge_mongodb/test/emqx_bridge_mongodb_SUITE.erl:533]: It's only used here twice, so I don't think it's worth doing it at this point.  If it starts getting used more often, it's a good idea.

**Comment by zmstone** [comment]
is it still needed after the client is fixed?

**Comment by thalesmg** [comment]
> is it still needed after the client is fixed?

I think it is better to have the option to customize it if needed, even if the auto mechanism is improved.

> bump mongo client version ?

It is bumped:

https://github.com/emqx/emqx/pull/11429/files#diff-f6aee252daac224fc006c3c07b8dfe64848a155b6150153d00678a3a273894bcR6

**Review by zhongwencool — APPROVED** [review]

**Pull request merged**